File: easy/minimumWaitingTime.cpp
#include <algorithm>
#include <iostream>
#include <vector>

int minimumWaitingTime(std::vector<int> waitingTimes);

int main() {
    std::vector<int> waitingTimes{4, 3, 6, 1};
    int minimumTime{minimumWaitingTime(waitingTimes)};

    std::cout << minimumTime << std::endl;

    return 0;
}

int minimumWaitingTime(std::vector<int> waitingTimes) {
    sort(waitingTimes.begin(), waitingTimes.end());

    int accumulate{0};
    int totalWaitingTime{0};
    for (int index{1}; index < static_cast<int>(waitingTimes.size()); ++index) {
        accumulate += waitingTimes[index - 1];
        totalWaitingTime += accumulate;
    }

    return totalWaitingTime;
}

File: easy/minimumWaitingTime.py
def minimum_waiting_time(array):
    array.sort()
    total_waiting_time = 0
    accumulate_times = 0
    for index in range(1, len(array)):
        accumulate_times += array[index - 1]
        total_waiting_time += accumulate_times

    return total_waiting_time


def main():
    times = [4, 3, 6, 1]
    print(minimum_waiting_time(times))


if __name__ == "__main__":
    main()
